scripts/splunk-to-segment.sh

```diff
@@ -85,17 +85,23 @@ function event_subject_map() {
   }'
 }
 
+# Not all event resources have a userId field necessary for attribution in Segment.
+# In such cases, the owner of the workspace is used instead.
+# For this to work workspaces must be named after a valid SSO username.
 jq \
   --compact-output \
   --slurpfile uidm <(get-uid-map.sh) \
+  --slurpfile wksm <(get-workspace-map.sh) \
   --slurpfile evvm <(event_verb_map) \
   --slurpfile evsm <(event_subject_map) \
-  '.result | select($uidm[0][.userId])
+  '.result
+  | .userIdOrWorkspace = (.userId // $wksm[0][.namespace])
+  | select(.userIdOrWorkspace and $uidm[0][.userIdOrWorkspace])
```

Comment:
Instead of doing the lookup in `$uidm` twice, might as well do that once here, place in a variable and then select on that variable.

Reply:
Done.